inst/www/shared/shiny.js

```diff
@@ -1260,106 +1279,801 @@
+          // and return.
+          if (brush.down.x === brush.up.x && brush.down.y === brush.up.y) {
+            brush.reset();
+            brushInfoSender.immediateCall();
+            return;
+          }
+
+          // Send info immediately on mouseup, but only if needed. If we don't
+          // do the pending check, we might send the same data twice (with
+          // with difference nonce).
+          if (brushInfoSender.isPending())
+            brushInfoSender.immediateCall();
+        }
+
+        function mouseupDragging(e) {
+          // Listen for left mouse button only
+          if (e.which !== 1) return;
+
+          $(document)
+            .off('mousemove.image_brush')
+            .off('mouseup.image_brush');
+
+          brush.up = mouseOffset(e);
+
+          brush.stopDragging();
+          setCursorStyle('grabbable');
+
+          if (brushInfoSender.isPending())
+            brushInfoSender.immediateCall();
+        }
+
+        // This should be called when the img (not the el) is removed
+        function onRemoveImg() {
+          if (opts.brushResetOnNew) {
+            brush.reset();
+            brushInfoSender.immediateCall();
+          }
+        }
+
+        if (!opts.brushResetOnNew) {
+          brush.importOldBrush();
+        }
+
+        return {
+          mousedown: mousedown,
+          mousemove: mousemove,
+          onRemoveImg: onRemoveImg
+        };
+      }
+
+
+      // We need to make sure that, if there's a dblClick listener, that a
+      // double-click doesn't trigger two click events. We'll trigger custom
+      // mousedown2 and dblclick2 events with this mousedown listener.
+      var clickCount = 0;
+      var clickTimer = null;
+      $el.on('mousedown.image_output', function(e) {
+        // Listen for left mouse button only
+        if (e.which !== 1) return;
+
+        // Information extracted from e that will get passed to the triggered
+        // event.
+        var eInfo = {
+          which:   e.which,
+          pageX:   e.pageX,
+          pageY:   e.pageY,
+          offsetX: e.offsetX,
+          offsetY: e.offsetY
+        };
+        var e2;
+
+        // If no dblclick listener, immediately trigger a click2 event.
+        if (!opts.dblclickId) {
+          e2 = $.Event('mousedown2', eInfo);
+          $el.trigger(e2);
+          return;
+        }
+
+        // If there's a dblclick listener, make sure not to count this as a
+        // click on the first mousedown; we need to wait for the dblclick
+        // delay before we can be sure this click was a single-click.
+        clickCount++;
+        if (clickCount === 1) {
+          clickTimer = setTimeout(function() {
+            e2 = $.Event('mousedown2', eInfo);
+            $el.trigger(e2);
+            clickCount = 0;
+
+          }, opts.dblclickDelay);
+
+        } else {
+          // If second click happens within specified delay, trigger our
+          // custom 'dblclick2' event.
```

Comment:
I would make sure that the mouse has not moved (or not beyond some very small number of pixels). If the user is quickly clicking lots of distinct data points you certainly don't want to interpret that as double-click.